sql/testdata/update

```diff
@@ -7,6 +7,16 @@ CREATE TABLE kv (
 statement ok
 INSERT INTO kv VALUES (1, 2), (3, 4), (5, 6), (7, 8)
 
+# traceon trace.out
+
+statement ok
+UPDATE kv SET v = k + v WHERE k = 3
+
+# traceoff
+
+statement ok
+UPDATE kv SET v = 4 WHERE k = 3
```

Comment:
Rather than do this, perhaps we should make a separate `tracedata` directory containing logic tests scripts that are not run by default but that can be run individually using `go test -run TestLogic -d tracedata/<foo>`.